okhttp/src/main/kotlin/okhttp3/internal/connection/RealConnectionPool.kt

```diff
@@ -357,35 +408,33 @@ class RealConnectionPool(
    * If there are already enough connections, we're done.
    * If not, we create one and then schedule the task to run again immediately.
    */
-  private fun ensureMinimumConnections(state: MinimumConnectionState): Long {
+  private fun openConnections(state: AddressState): Long {
     // This policy does not require minimum connections, don't run again
-    if (state.policy.minimumConcurrentCalls < 1) return -1
-
-    var unsatisfiedCountMinTask = state.policy.minimumConcurrentCalls
+    if (state.policy.minimumConcurrentCalls == 0) return -1L
 
+    var concurrentCallCapacity = 0
     for (connection in connections) {
+      if (state.address != connection.route.address) continue
```

Comment:
To be honest I just felt like a boring equality check was simpler.

On further reflection, I think it’s the right call.

`isEligible()` relies on environment-dependent heuristics. We have code to recover if we get an HTTP 421 because we guessed wrong. But if we guess wrong here, we don’t validate that.

In theory coalescing could cause us to double count. If the user asks for 32 concurrent calls to apis.squareup.com plus 32 concurrent calls to cdn.squareup.com, we could count a single 32-stream HTTP/2 connection against both.